modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/service/impl/CommercePaymentEntryLocalServiceImpl.java

```diff
@@ -434,6 +466,38 @@ else if (!reasonKey.equals(commercePaymentEntry.getReasonKey())) {
 		return commercePaymentEntryPersistence.update(commercePaymentEntry);
 	}
 
+	private Predicate _getPredicate(
+		long companyId, long classNameId, long classPK) {
+
+		return CommercePaymentEntryTable.INSTANCE.classPK.in(
+			DSLQueryFactoryUtil.select(
+				CommercePaymentEntryTable.INSTANCE.commercePaymentEntryId
+			).from(
+				CommercePaymentEntryTable.INSTANCE
+			).where(
+				CommercePaymentEntryTable.INSTANCE.companyId.eq(
+					companyId
+				).and(
+					CommercePaymentEntryTable.INSTANCE.classNameId.eq(
+						classNameId)
+				).and(
+					CommercePaymentEntryTable.INSTANCE.classPK.eq(
+						classPK
+					).and(
+						CommercePaymentEntryTable.INSTANCE.paymentStatus.eq(
+							CommercePaymentEntryConstants.STATUS_COMPLETED)
+					)
+				).and(
```

Comment:
Formatting of this seems off.
I think this is what the query would look like if translated,

`select commercePaymentEntryId from CommercePaymentEntry where companyId = 123 and classNameId = 123 and (classPK = 123 and paymentStatus = 123) and type = 123;`

I dont think we need to add the parenthesis since its all "and"?